From 8f70a049daa30be894158411439a36f920f0d11c Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 15 Jan 2014 20:13:04 +0100
Subject: [PATCH] sun5i: Tweak pll6 init value

There are multiple ways to get 600 MHz from PLL6, the sun5i ccmu is very
similar to the sun4i and sun7i ccmu.

This commit changes the PLL6 initialization we do for sun5i to make the PLL6
reg value match that of sun4i and sun7i. This also makes it closer to the
sun5i power on default, as we are now only changing the K-factor from 3 to 1.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 arch/arm/cpu/armv7/sunxi/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/cpu/armv7/sunxi/clock.c b/arch/arm/cpu/armv7/sunxi/clock.c
index b9dd608..54b8753 100644
--- a/arch/arm/cpu/armv7/sunxi/clock.c
+++ b/arch/arm/cpu/armv7/sunxi/clock.c
@@ -46,7 +46,7 @@ static void clock_init_safe(void)
 #ifdef CONFIG_SUN5I
 	/* Power on reset default for PLL6 is 2400 MHz, which is faster then
 	 * it can reliable do :|  Set it to a 600 MHz instead. */
-	writel(0x21009900, &ccm->pll6_cfg);
+	writel(0x21009911, &ccm->pll6_cfg);
 #endif
 #ifdef CONFIG_SUN7I
 	writel(0x1 << 6 | readl(&ccm->ahb_gate0), &ccm->ahb_gate0);
-- 
1.8.5.1

